frontend/pages/browse.js

```diff
@@ -123,46 +127,75 @@ export default function Index({
     if (params.message) setMessage(decodeURI(params.message));
     if (params.errorMessage) setErrorMessage(decodeURI(params.errorMessage));
   });
+
   const [tabValue, setTabValue] = React.useState(hash ? typesByTabValue.indexOf(hash) : 0);
   const [filtersExpanded, setFiltersExpanded] = React.useState(false);
+
   const [filters, setFilters] = React.useState({
     projects: {},
     members: {},
     organizations: {},
   });
 
+  /**
+   * When applying new filters, update the state of the
+   * loading spinner until the async call finishes.
+   */
   const applyNewFilters = async (type, newFilters, closeFilters) => {
-    if (filters !== newFilters) {
```

Comment:
Instead of grouping everything within this conditional, I refactored it outside and added an early short circuit conditional if the objects are equal.

@ reviewers, wdyt? We could consider [isEqual](https://www.ecosia.org/search?q=lodash+is+equal&addon=chrome&addonversion=3.3.0&method=topbar) for object comparison if desired.